csrc/host_ir/container.h

```diff
@@ -0,0 +1,49 @@
+// clang-format off
+/*
+ * SPDX-FileCopyrightText: Copyright (c) 2023-present NVIDIA CORPORATION & AFFILIATES.
+ * All rights reserved.
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+// clang-format on
+#pragma once
+
+#include <fusion.h>
+
+namespace nvfuser {
+
+namespace hir {
+
+/*
+HostIrContainer is used to represent a host program.
+1) It inherits from Fusion, so that (Host) IRs can be resgistered to it.
```

Comment:
Registering happens under the hood when we create an IR. `HostIrContainer` can be passed to `IrBuilder` because it inherits from `Fusion` and thus `IrContainer`

Reply:
I was asking whether HostIrContainer can inherit from IrContainer but not Fusion. I failed to see how that would break.

Reply:
> I was asking whether HostIrContainer can inherit from IrContainer but not Fusion. I failed to see how that would break.

Oh, ok, sorry I didn't understand that was your suggestion. I have no problem making it inherit from `IrContainer` rather than `Fusion` at the moment, so let me fix that.

The main reason I made it inherit from `Fusion` was
1) to mimick what is done with `kir::Kernel` (as pointed out by @naoyam)
2) to later be able to use utils (such as `IterVisitor`) which are for the most part implemented for Fusion (however imo they should be implemented for `IrContainer`)

Reply:
Sorry, while I am trying to compile the program with `HostIrContainer` inheriting from `IrContainer` and not `Fusion`, I'm realizing I'm using the following `Fusion`'s features:
1) `addInput`, `addOutput`, `inputs`, `outputs`
2) `FusionGuard::setCurFusion(Fusion*)`

So I'd rather stick with Fusion for now but Im happy to revisit later. Let me know what you think